test/e2e/e2e_test.go

```diff
@@ -148,6 +148,31 @@ var _ = Describe("Workload cluster creation", func() {
 				ControlPlane: result.ControlPlane,
 			}, e2eConfig.GetIntervals(specName, "wait-control-plane")...)
 
+			By("Upgrading control plane")
+			ApplyClusterTemplateAndWait(ctx, ApplyClusterTemplateAndWaitInput{
```

Comment:
I don't think this waits for the upgrade to have completed.

Reply:
good point, I checked `WaitForControlPlaneToBeReady` function below, it's going to return true if rollout wasn't started immediately. I will add a new one `WaitForControlPlaneToUpgrade` that wait for all machines to machines to have updated spec